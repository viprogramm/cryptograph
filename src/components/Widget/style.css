body{
    font-size:18px;
}
.fell {
    color: #a22921;
}

.up {
    color: #61c27b;
}

.svg{
    border: 1px solid gray;
}