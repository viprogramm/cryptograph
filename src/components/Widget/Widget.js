import React from "react";
import { string, array, number } from "prop-types";
import Label from "../Label/Label";
import Rate from "../Rate/Rate";
import Graph from "../Graph/Graph";

import "./style.css";

const Widget = props => {
  const { coin, data, count } = props;
  return (
    <div>
      <Label coin={coin} currency="USD" />
      <Rate rates={data} />
      <Graph className="svg" data={data} count={count} />
    </div>
  );
};

Widget.propTypes = {
  coin: string.isRequired,
  data: array.isRequired,
  count: number.isRequired
};

export default Widget;
