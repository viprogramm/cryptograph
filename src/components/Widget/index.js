import withWebSocket from "../../hocs/withWebSocket";
import withMapDataByProp from "../../hocs/withMap/dataByProp";
import Widget from "./Widget";

export default withWebSocket(withMapDataByProp("price")(Widget));
