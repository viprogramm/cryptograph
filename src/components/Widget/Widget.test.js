import React from "react";
import renderer from "react-test-renderer";
import Widget from "./Widget";

describe("Widget", () => {
  test("withu props", () => {
    const tree = renderer
      .create(<Widget coin="BTC" data={[10, 20, 30]} count={3} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
