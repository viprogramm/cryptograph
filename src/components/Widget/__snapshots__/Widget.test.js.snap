// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Widget withu props 1`] = `
<div>
  <h3>
    BTC - USD
  </h3>
  <div
    className="rates"
  >
    <span
      className="up rate"
    >
      $
      30
    </span>
     
    <span
      className="up diff"
    >
      (
      200
      %)
    </span>
  </div>
  <svg
    className="svg"
    height={100}
    width={220}
  >
    <path
      d="M 0 100 L 0 0 L 110 50 L 220 100 L 220 100 z"
      fill="#daa49d"
      stroke="#a22921"
      strokeWidth={2}
    />
  </svg>
</div>
`;
